seatunnel-engine/seatunnel-engine-storage/imap-storage-api/pom.xml

```diff
@@ -31,6 +31,12 @@
     <artifactId>imap-storage-api</artifactId>
 
     <dependencies>
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-api</artifactId>
+            <version>${project.version}</version>
+        </dependency>
```

Comment:
As a base module, I don't think we need to depend on seatunnel-api, how confusing it is.

Reply:
> As a base module, I don't think we need to depend on seatunnel-api, how confusing it is.

'IMapStorageFactory' implements' org.apache.seatunnel.api.table.factory.Factory', Whether we need to move the Factory here?

Reply:
It's messy, it's a connector thing (`seatunnel-api`), I don't like mixing things together, it's important to keep the established boundaries.

Reply:
I have some doubts about this, so I think it is necessary to discuss here.Now the SPI obtains the Factory mainly for the connector,Should we migrate this function to `seatunnel-common` or implement a lightweight function in the engine?
Please give me some advice, thanks!

@CalvinKirs @EricJoy2048 @ashulin @hailin0 @TyrantLucifer @Hisoka-X @ic4y .

Reply:
Seem like can't use Factory which in `seatunnel-api`, because it need return option rule which design for connector, Just create `IMapStorageFactory` without implement Factory in `seatunnel-api` is fine for me.

Reply:
> 

If we don't implement Factory, then we need to implement a FactoryUtil. Can I understand and implement a set of lightweight SPI functions?